tests/server/core/hot_region_storage_test.go

```diff
@@ -0,0 +1,145 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	"github.com/tikv/pd/server"
+	"github.com/tikv/pd/server/config"
+	"github.com/tikv/pd/server/core"
+	"github.com/tikv/pd/server/statistics"
+	"github.com/tikv/pd/tests"
+	"github.com/tikv/pd/tests/pdctl"
+)
+
+func Test(t *testing.T) {
+	TestingT(t)
+}
+
+var _ = Suite(&hotRegionHistorySuite{})
+
+type hotRegionHistorySuite struct{}
+
+func (s *hotRegionHistorySuite) SetUpSuite(c *C) {
+	server.EnableZap = true
+}
+
+func (s *hotRegionHistorySuite) TestHotRegionStorage(c *C) {
+	statistics.Denoising = false
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	cluster, err := tests.NewTestCluster(ctx, 1,
+		func(cfg *config.Config, serverName string) {
+			cfg.Schedule.HotRegionCacheHitsThreshold = 0
+			cfg.Schedule.HotRegionsWriteInterval.Duration = 1000 * time.Millisecond
+			cfg.Schedule.HotRegionsResevervedDays = 1
+		},
+	)
+	c.Assert(err, IsNil)
+	err = cluster.RunInitialServers()
+	c.Assert(err, IsNil)
+	cluster.WaitLeader()
+	stores := []*metapb.Store{
+		{
+			Id:            1,
+			State:         metapb.StoreState_Up,
+			LastHeartbeat: time.Now().UnixNano(),
+		},
+		{
+			Id:            2,
+			State:         metapb.StoreState_Up,
+			LastHeartbeat: time.Now().UnixNano(),
+		},
+	}
+
+	leaderServer := cluster.GetServer(cluster.GetLeader())
+	c.Assert(leaderServer.BootstrapCluster(), IsNil)
+	for _, store := range stores {
+		pdctl.MustPutStore(c, leaderServer.GetServer(), store)
+	}
+	defer cluster.Destroy()
+	startTime := time.Now().UnixNano() / int64(time.Millisecond)
+	pdctl.MustPutRegion(c, cluster, 1, 1, []byte("a"), []byte("b"), core.SetWrittenBytes(3000000000), core.SetReportInterval(statistics.WriteReportInterval))
+	pdctl.MustPutRegion(c, cluster, 2, 2, []byte("c"), []byte("d"), core.SetWrittenBytes(6000000000), core.SetReportInterval(statistics.WriteReportInterval))
+	pdctl.MustPutRegion(c, cluster, 3, 1, []byte("e"), []byte("f"))
+	pdctl.MustPutRegion(c, cluster, 4, 2, []byte("g"), []byte("h"))
+	storeStatss := []*pdpb.StoreStats{
+		{
+			StoreId:  1,
+			Interval: &pdpb.TimeInterval{StartTimestamp: 0, EndTimestamp: statistics.ReadReportInterval},
+			PeerStats: []*pdpb.PeerStat{
+				{
+					RegionId:  3,
+					ReadBytes: 9000000000,
+				},
+			},
+		},
+		{
+			StoreId:  2,
+			Interval: &pdpb.TimeInterval{StartTimestamp: 0, EndTimestamp: statistics.ReadReportInterval},
+			PeerStats: []*pdpb.PeerStat{
+				{
+					RegionId:  4,
+					ReadBytes: 9000000000,
+				},
+			},
+		},
+	}
+	for _, storeStats := range storeStatss {
+		leaderServer.GetRaftCluster().HandleStoreHeartbeat(storeStats)
+	}
+	// wait hot scheduler starts
+	time.Sleep(5000 * time.Millisecond)
```

Comment:
It's too long for a test,  can it be 200ms?

Reply:
other test in hot test also set sleep time as 5000*time.Millsecond

Reply:
other test like hot_test set time to 5000*time.Millsecond.It’s ok on my computer, but I don’t know if it’s not working on computers with worse performance.

Reply:
I think it's ok. `hot_test` should be too long

Reply:
I think it will decrease the test case stability even it sleep 5000ms  , may add finish signal to solve it.   
And it will add IT if one test case running time cost more than 1s.

Reply:
what`s mean  to add  finish signal?

Reply:
let's change to 200 millisecond.

Reply:
sorry,I tried  again,if we set time less than 5000 milliseconds,it can to be done